README.md

```diff
@@ -102,6 +103,22 @@ with slurmdbdconfig.edit("/etc/slurm/slurmdbd.conf") as config:
     del config.auth_alt_parameters
 ```
 
+##### `cgroupconfig`
+
+This module provides an API for editing _cgroup.conf_ files, and create new _cgroup.conf_
+files if they do not exist. Here's some operations you can perform on the _cgroup.conf_
+file using this editor:
+
+```python
+from slurmutils.editors import cgroupconfig
+
+with cgroupconfig.edit("/etc/slurm/cgroup.conf") as config:
+    config.constrain_cores = "yes"
+    config.constrain_devices = "yes"
+    config.constrain_ram_space = "yes"
+    config.constrain_swap_space = "yes"
```

Comment:
Thought: reading this again, maybe we should also allow setting the configs using booleans? It doesn't appear to be too hard to allow setting it with "yes/no" and with True/False.

Reply:
That's something we could have as an enhancement in the future. Early versions of the editors - _back when they were just a standalone charm library_ - supported converting back and forth between formats (Yes <-> True & No <-> False), but I didn't carry over the functionality in the interest of time :sweat_smile: 

Let me open an issue with the enhancement tag. Probably a good first issue too since you just need to implement the callback in the `callback` module.

Reply:
Created a new issue to track this enhancement: https://github.com/charmed-hpc/slurmutils/issues/22 

Eventually I would like to get where we can lint the data models before the respective configuration files are updated because then we can catch any problems early on rather than when Slurm explodes and we're sad.